### PR TITLE
fix: intermittent failures in governance tests with tsan

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
Intermittent failure in functional tests feature_governance and feature_governance_cl:
 - https://gitlab.com/dashpay/dash/-/jobs/7574959798
 - https://gitlab.com/dashpay/dash/-/jobs/7574917439
 - https://gitlab.com/dashpay/dash/-/jobs/7576050419

and many other

## What was done?
Bump timeout in `feature_governance.py`, `feature_governance_cl.py`; moved code from them to `test_framework/governance.py`

## How Has This Been Tested?
Run unit/functional tests in 40 threads locally: amount of failures decreased from 50% to 2.5%
```
test/functional/test_runner.py -j40  feature_governance_cl.py feature_governance_cl.py feature_governance_cl.py feature_governance_cl.py feature_governance_cl.py feature_governance_cl.py feature_governance_cl.py feature_governance_cl.py feature_governance_cl.py feature_governance_cl.py feature_governance.py feature_governance.py feature_governance.py feature_governance.py feature_governance.py feature_governance.py feature_governance.py feature_governance.py feature_governance.py feature_governance_cl.py feature_governance_cl.py feature_governance_cl.py feature_governance_cl.py feature_governance_cl.py feature_governance_cl.py feature_governance_cl.py feature_governance_cl.py feature_governance_cl.py feature_governance_cl.py feature_governance.py feature_governance.py feature_governance.py feature_governance.py feature_governance.py feature_governance.py feature_governance.py feature_governance.py feature_governance.py
```

## Breaking Changes
N/A

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: test/functional/test_framework/governance.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018-2024 The Dash Core developers
```
Comment (PastaPastaPasta) [inline test/functional/test_framework/governance.py:2]:
```suggestion
# Copyright (c) 2024 The Dash Core developers
```

Comment (knst) [reply test/functional/test_framework/governance.py:2]: that's an old code moved from other file

Comment (PastaPastaPasta) [reply test/functional/test_framework/governance.py:2]: 🤷‍♂️

**Review by PastaPastaPasta — APPROVED** [review]
utACK 291716a8b4580261a93b4f8dd5b431ed79620aeb

**Review by UdjinM6 — APPROVED** [review]
utACK 291716a8b4580261a93b4f8dd5b431ed79620aeb

**Pull request merged**